datadog_lambda/tracing.py

```diff
@@ -380,10 +381,26 @@ def extract_context_from_step_functions(event, lambda_context):
         execution_id = event.get("Execution").get("Id")
         state_name = event.get("State").get("Name")
         state_entered_time = event.get("State").get("EnteredTime")
-        # returning 128 bits since 128bit traceId will be break up into
-        # traditional traceId and _dd.p.tid tag
-        # https://github.com/DataDog/dd-trace-py/blob/3e34d21cb9b5e1916e549047158cb119317b96ab/ddtrace/propagation/http.py#L232-L240
-        trace_id = _deterministic_sha256_hash(execution_id, LOWER_64_BITS)
+        meta = {}
+
+        if "_datadog" in event:
+            # use the trace ID from the top-most parent when it exists
+            trace_header = event.get("_datadog")
+            trace_id = trace_header.get(TraceHeader.TRACE_ID)
+            tags = trace_header.get(TraceHeader.TAGS)
+            for tag in tags.split(","):
```

Comment:
This way we pass all the tags and not just `_dd.p.tid`, not sure if there's any unintended consequences